antarest/study/storage/variantstudy/model/command/create_st_storage.py

```diff
@@ -0,0 +1,381 @@
+import json
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
+
+import numpy as np
+from antarest.core.model import JSON
+from antarest.matrixstore.model import MatrixData
+from antarest.study.storage.rawstudy.model.filesystem.config.model import (
+    Area,
+    FileStudyTreeConfig,
+)
+from antarest.study.storage.rawstudy.model.filesystem.config.st_storage import (
+    STStorageConfig,
+)
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.variantstudy.business.matrix_constants_generator import (
+    GeneratorMatrixConstants,
+)
+from antarest.study.storage.variantstudy.business.utils import (
+    strip_matrix_protocol,
+    validate_matrix,
+)
+from antarest.study.storage.variantstudy.model.command.common import (
+    CommandName,
+    CommandOutput,
+)
+from antarest.study.storage.variantstudy.model.command.icommand import (
+    MATCH_SIGNATURE_SEPARATOR,
+    ICommand,
+)
+from antarest.study.storage.variantstudy.model.model import CommandDTO
+from pydantic import Field, validator, Extra
+from pydantic.fields import ModelField
+
+# noinspection SpellCheckingInspection
+_MATRIX_NAMES = (
+    "pmax_injection",
+    "pmax_withdrawal",
+    "lower_rule_curve",
+    "upper_rule_curve",
+    "inflows",
+)
+
+# Minimum required version.
+REQUIRED_VERSION = 860
+
+MatrixType = List[List[MatrixData]]
+
+
+# noinspection SpellCheckingInspection
+class CreateSTStorage(ICommand):
+    """
+    Command used to create a short-terme storage in an area.
+    """
+
+    class Config:
+        extra = Extra.forbid
+
+    # Overloaded parameters
+    # =====================
+
+    command_name = CommandName.CREATE_ST_STORAGE
+    version = 1
+
+    # Command parameters
+    # ==================
+
+    area_id: str = Field(description="Area ID", regex=r"[a-z0-9_(),& -]+")
+    parameters: STStorageConfig
```

Comment:
Although it's important to introduce a stronger model than in existing commands, it will be a design issue in the long run to use the same model all the way from file storage to API.

We will need to separate layers at some point. We should probably have at least 2 layers, maybe 3:
- one for the file format
- one for the business model (in-memory representation)
- possibly one more for the API (DTOs)

This separation will:
- be needed when we want to have another storage strategy than the current file format (database, other format ...).
- allow us to have a better user facing model, without being constrained by the current file format.

Reply:
The separation between the file format and the business model is a hard work, but we (with Florian) have some good ideas for the future.

An effort is done here to separate the File format and the DTO. But again, this is a work in progress…